lib/kintsugi/apply_change_to_project.rb

```diff
@@ -148,13 +152,17 @@ def apply_file_changes(project, additions, removals)
       end
 
       file_removals.each do |change, path|
-        next unless addition_to_paths[change].nil?
+        next unless addition_keys_to_paths[apply_file_changes_change_key(change)].nil?
 
         file_reference = project[join_path(path, change["displayName"])]
         remove_component(file_reference, change)
       end
     end
 
+    def apply_file_changes_change_key(change)
```

Comment:
If it's only going to be used in the function, move it into the function and remove the prefix from the name